pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -150,6 +152,18 @@ func createRouteStatus(parentResults []RouteParentResult, obj config.Config, cur
 			}
 		}
 
+		// when ambient is enabled, report the waypoints resolved condition for HTTPRoute types
+		if features.EnableAmbient && obj.GroupVersionKind.Kind == gvk.HTTPRoute.Kind {
```

Comment:
This actually applies to all route types -- even TCPRoute

Reply:
Actually technically TCPRoute doesn't work in ztunnel _or_ waypoint right now, but https://github.com/istio/istio/pull/55198 fixes that.

Reply:
yea that's true, i'll only gate for the ambient enabled feature.